synapse/http/endpoint.py

```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from twisted.internet.endpoints import SSL4ClientEndpoint, TCP4ClientEndpoint
+from twisted.internet.endpoints import HostnameEndpoint, wrapClientTLS
```

Comment:
Gotcha; yeah.  I'd be inclined to eliminate the default, personally; I find that default args are useful a lot more rarely than they _seem_ like they'd be useful :)